sasdata/data_util/manipulations.py

```diff
@@ -944,6 +946,10 @@ def _agv(self, data2D, run='phi'):
 
             # Get the binning index
             if run.lower() == 'phi':
+                if phi_value < binning.min:
+                    phi_value += 2 * np.pi
+                elif phi_value > binning.max:
+                    phi_value -= 2 * np.pi
```

Comment:
Are you sure this is correct?  I think this should be omitted as I understand the math? Maybe we can discuss? I could be doing the maths wrong. As @lucas-wilkins points out : "angles are the WORST" 

my thinking:
for example if phi_min is 320 degrees, phi_value is 340 and phi_max (on the other side of 0) is 20.  In this case phi_value is > phi_min which is correct. But it is also > phi_max.  I don't see why you need to subtrac 2pi which just makes it a negative angle ... and would now make phi_value < phi_min which violates condition one?

Reply:
It must be correct. Removing these lines causes the bug to return (it's fixed when the central line is above the x-axis, but still broken when below it). `self.phi_min` and `self.phi_max` aren't restricted to the 0 - 2pi range, and the check on line 884 ensures self.phi_max is always numerically greater than self.phi_min. Therefore, if `phi_value` > `binning.max` (which is the same as `self.phi_max`), then this subtraction has to take place or `phi_value` will be outside the allowed range and `get_bin_index` will return a value > `self.nbins`.

Reply:
Now I am very (very) confused. Of course not failing <> from correct, but still, I cannot even see how that second bit of code ever gets used since phi_value at that point should always be smaller than phi_max no matter what (If we've straddled zero phi_max should be >2PI while phi_value should not be)? Unless wedge_slicer is sending something over I don't understand? But if the code fails without that `elif` statement then it clearly is hitting that code. Going to have to crack my geometry books here.

Reply:
Right.... so we may both be wrong. I'm starting to feel like an idiot 😒 .  I think I'm beginning to see why they created `flip_phi`. I believe that most of the problem goes away without adding these tests, if done right. But there is still a few ends I've not quite finished figuring out which I think come from `WedgeInteractor` maybe?  @lucas-wilkins is right: angles are a real pain 😕.

Will continue tomorrow and hopefully post my findings.

Reply:
> we may both be wrong

I'm don't actually understand what the problem is here. There must be something you've got in mind which hasn't occurred to me, because as far as I can see this code is fine.

Reply:
Ok @lucas-wilkins and @ehewins, I’ve now done what I had avoided doing so far: really understand how the sector class works in `manipulation.py`.  Changes needed are of course minor but not as simplistic as I’d originally thought ☹️. The really tricky/confusing bit is that there are like 4(?) Different angular range spaces (-pi, pi; 0,2pi; -delta, 2pi-delta; 0+delta,2pi+delta). `phi_val` and `phi_max/min` are in the 0, 2pi interval. However `self.phi_min/max` (yes `self.phi` is different from `phi` ☹️) is unknown and can be in any of the last 3 ranges.  The gui slicer is in the -pi to pi range natively.
Trying to understand it all I ended up writing my own version of manipulations.py changes which is very close to  yours @ehewins , but not the same. I have created a new branch off of your branch (called v3) as it was the easiest way I could think of to show what I did and why -- which also means it has extensive comments which would be good to capture somewhere 🙂 The difference is that, having figured out the subtle differences in angular intervals, I tried to keep them separate AND not change that range like I originally did (e.g. using phi_max += 2* np.py) to try to keep what I was doing clean and clear (else I confuse myself). 
Now that I understand it, I believe your code also is correct. The main difference is that you are mixing `phi_value` (in 0, 2pi space) with `self.phi_min/max` (=`binning.min/max`) which can be in any of 3 spaces (in fact 2 I think since you actually altered `self.phi_max`).  That is why you ended up with the `if elif` that confused me. They are both required in order to deal both with the straddling problem and the fact that `phi_value` might be in a different angular space. The other difference is just a preference of starting point.  My plot in sasview gui doesn’t go negative till phi_min is below the x axis (so the plot can go more than pi but never less than -pi) while your version will start negative when the central line crosses the x axis meaning that the plot can go to less than -pi but not more than pi.
To answer @lucas-wilkins  point about tests, I don’t know of any good test to verify that we are including all the points we are supposed to from the ROI, but we can do something close if we had some rocking curve of flux line data I think. We also would need to have tests for each of 3 cases: not straddling the discontinuity, straddling but with the central line above the x axis and the central line below the x axis.
While it is really important to get some good unit tests, I think that should not stop this PR as long as a new issue is put in for that?